lib/ramble/ramble/workspace/workspace.py

```diff
@@ -1608,6 +1608,11 @@ def shared_license_dir(self):
         """Path to the shared license directory"""
         return os.path.join(self.shared_dir, workspace_shared_license_path)
 
+    def package_manager_dir(self, package_manager):
+        if package_manager is not None:
+            return os.path.join(self.software_dir, package_manager.name)
+        return os.path.join(self.software_dir, "no-package-manager")
```

Comment:
Do you think when `package_manager` is the "no package manager" package manager it should still have a valid object to give a more consistent interface?

Reply:
I don't dislike the idea, but I also can't decide what is a reasonable name for that package manager.

TBF, I think this case should *never* happen (because if the package manager is none, then it will never actually try to create a software environment, and it won't need to have it's own directory).

So this is more a very unnecessary fallback mode to avoid some gross error from popping up if you call this method without a package manager.

Reply:
I updated this so that the `package_manager_dir` is an attribute of the package manager, rather than the workspace. This should get around the error case where `package_manager` is None now to help simplify the interface.